modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2Utils.java

```diff
@@ -569,13 +588,28 @@ public static boolean checkAndStartNotStartedCache(GridKernalContext ctx, GridH2
         return false;
     }
 
+    /**
+     * @param cls The class whose convertibility is to be tested.
+     * @param colType Column target type.
+     * @return Whether specified class can be implicitly converted to the specified type.
+     * @see #wrap(CacheObjectValueContext, Object, int)
+     */
+    public static boolean isConvertableToColumnType(Class<?> cls, int colType) {
```

Comment:
I don't like the method name, it assumes that `true` should be returned also in case the class exactly match the column type. But, for example, it will return `false` for `java.sql.Timestamp`/`TIMESTAMP` pair.

Reply:
I think it worth to support the case you described.  May be it is out of the ticket's scope but it makes the logic more clear. I have added the additional check if the class exactly matches the column type.

Reply:
But now default `return false` implementation looks illogical for indexing. Perhaps we shouldn't provide default implementation for this method. WDYT? I doubt anyone have indexing implementation other than ignite-indexing, so we should not worried about backward compatibility (at least, new non-default methods were added recently and nobody complained)

Reply:
It seems that you are right. I removed the default implementation.